include/bl_storage.h

```diff
@@ -7,15 +7,59 @@
 #ifndef BL_STORAGE_H_
 #define BL_STORAGE_H_
 
-#include <zephyr/types.h>
 #include <string.h>
+#include <zephyr/types.h>
+#include <drivers/nrfx_common.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-
 #define EHASHFF 113 /* A hash contains too many 0xFs. */
+#define EREADLCS 114 /* LCS field of OTP is in an invalid state */
+#define EINVALIDLCS 115 /* Invalid LCS*/
+
+/* We truncate the 32 byte sha256 down to 16 bytes before storing it */
+#define SB_PUBLIC_KEY_HASH_LEN 16
+
+/** Storage for the PRoT Security Lifecycle state, that consists of 4 states:
+ *  - Device assembly and test
+ *  - PRoT Provisioning
+ *  - Secured
+ *  - Decommissioned
+ *  These states are transitioned top down during the life time of a device.
+ *  The Device assembly and test state is implicitly defined by checking if
+ *  the provisioning state wasn't entered yet.
+ *  This works as ASSEMBLY implies the OTP be erased.
```

Comment:
```suggestion
 *  This works as ASSEMBLY implies the OTP is erased.
```
might be a better way to write this.